Actions/AL-Go-Helper.ps1

```diff
@@ -553,6 +553,7 @@ function ReadSettings {
     $settings = [ordered]@{
         "type"                                          = "PTE"
         "unusedALGoSystemFiles"                         = @()
+        "updateALGoBranches"                            = @()
```

Comment:
Shouldn't this be as a parameter on Update AL-Go for GitHub as well?
Supporting a,b,c including masks - like *
and... - if not provided there we can grab the branches setting

Reply:
That's an idea.

But it might be confusing as there is already a branch field when dispatching the workflow:
![image](https://github.com/microsoft/AL-Go/assets/43066499/25309a02-2a6d-4113-8437-1b1ab5cbb33f)

I'm also contemplating the way the setting(s) should be structured. I feel like workflow-specific settings should have their own place and better structure. E.g. what if there was checkbox when dispatching the workflow **"Use values from settings"** and then somewhere there is an `UpdateALGoSettings`:
```
...
{
    supportedBranches: [
        { 
            branch: "main",
            createPR: false,
            pullLatest: true
       },
       { 
           branch: "releases/*",
           createPR: true
      }
    ],
    defaults: {
        createPR: false,
        pullLatest: true
    }
}
...
```

Reply:
I think it is overcomplicating things to have a structure and the problem with settings is that they need to be in one of the branches right (the one you select when you dispatch the workflow - probably main) and if you want to change if - you need to create a PR to do so.

If you select a different branch in workflow dispatch - you get the settings in that branch, which might be old/different.

Another idea would be to have a GitHub variable to specify this, but unfortunately, github actions syntax doesn't support:

```
inputs:
  ...
  default: ${{ vars.defaultvalue }}
```

default needs to be a hardcoded string.

I would suggest to not have this in settings - and only have an input variable, which (if not specified) takes it's value from a GitHub variable. And if you don't want to include the branches in the github variable, you can just specify a minus in includebranches.

Reply:
BTW Settings also have the problem that they are included when creating a fork - and the branches in that fork might be very different.

Reply:
Switched to using a GitHub variable.

Reply:
Reverted back to using using a setting.

"Update AL-Go System Files" can run on multiple branches only when on schedule. Manually triggering "Update AL-Go System Files" will run it only on the specified branch.